lib/spack/llnl/util/filesystem.py

```diff
@@ -1734,83 +1739,78 @@ def find(root, files, recursive=True, max_depth: Optional[int] = None):
     elif max_depth is None:
         max_depth = sys.maxsize
 
-    tty.debug(f"Find (max depth = {max_depth}): {root} {str(files)}")
+    tty.debug(f"Find (max depth = {max_depth}): {root} {files}")
     result = find_max_depth(root, files, max_depth)
-
-    tty.debug(f"Find complete: {root} {str(files)}")
+    tty.debug(f"Find complete: {root} {files}")
     return result
 
 
-@system_path_filter(arg_slice=slice(1))
-def find_max_depth(root, globs, max_depth: Optional[int] = None):
-    """Given a set of non-recursive glob file patterns, finds all
-    files matching those patterns up to a maximum specified depth.
+def _dir_id(s: os.stat_result) -> Tuple[int, int]:
+    # Note: on windows, st_ino is the file index and st_dev is the volume serial number. See
+    # https://github.com/python/cpython/blob/3.9/Python/fileutils.c
+    return (s.st_ino, s.st_dev)
```

Comment:
(minor) I originally defined this inside `find_max_depth` because it was only used there: IMO the purpose of this function is unclear confusing outside of that context (really it's just there to bring a reader to the comment).

Likewise for `_log_file_access_issue`

Reply:
It starts with a _

Reply:
I was aware it started with a `_` when I asked you to move this. If the `_` was your main reason for moving it, move it back.

Reply:
I disagree. For one in your original implemenation it captured a variable that was unbound at time of definition. You don't have those scoping issues here. Nobody read _* functions without checking where they are called from anyways.

Reply:
> For one in your original implemenation it captured a variable that was unbound at time of definition.

what do you mean by this precisely? if we have

```
    def _dir_id(stat_info):
        # Note: on windows, st_ino is the file index and st_dev
        # is the volume serial number. See
        # https://github.com/python/cpython/blob/3.9/Python/fileutils.c
        return (stat_info.st_ino, stat_info.st_dev)
```

then the only variable here is `stat_info`, and it's a parameter to the function, so I'm not clear what you mean.

Internal function definitions can be problematic when they accidentally reference defined variables (e.g. when the internal definition accidentally references a variable at different scope), but that's the only argument against internal definitions that I'm aware of (other than pickling considerations, which also don't apply here).